src/app/components/queue/form/QuestionForm.tsx

```diff
@@ -0,0 +1,209 @@
+import {
+  Box,
+  Button,
+  Checkbox,
+  Drawer,
+  FormControl,
+  FormControlLabel,
+  FormLabel,
+  TextField,
+} from "@mui/material";
+import { IoMdClose } from "react-icons/io";
+import React from "react";
+import { officeHourContext } from "@context/OfficeHourContext";
+import { MultipleChoiceTags, SingleChoiceTags } from "./Tags";
+import { TagOption } from "@interfaces/db";
+import { createQuestion } from "@utils/index";
+import { serverTimestamp } from "firebase/firestore";
+import { UserSessionContext } from "@context/UserSessionContext";
+import { IdentifiableQuestion } from "@interfaces/type";
+
+interface QuestionFormProps {
+  // button to open the form
+  triggerButton: JSX.Element;
+  title: string;
+  // current state of question
+  currentQuestion: IdentifiableQuestion;
+}
+
+const QuestionForm = (props: QuestionFormProps) => {
+  const { triggerButton, title, currentQuestion } = props;
+  const [openForm, setOpenForm] = React.useState(false);
+  const ohContext = React.useContext(officeHourContext);
+  const userSessionContext = React.useContext(UserSessionContext);
+
+  const [question, setQuestion] = React.useState(currentQuestion.title);
+  const [description, setDescription] = React.useState(
+    currentQuestion.description
+  );
+  const [questionPublic, setQuestionPublic] = React.useState(
+    currentQuestion.questionPublic
+  );
+
+  const [questionTags, setQuestionTags] = React.useState<
+    Record<string, TagOption[]>
+  >(() => {
+    const curr_tags = currentQuestion.tags.map((t) => t.choice);
+    let init: Record<string, TagOption[]> = {};
+    Object.keys(ohContext.course.tags).forEach((t) => {
+      init[t] = ohContext.course.tags[t].options.filter((o) =>
+        curr_tags.includes(o.choice)
+      );
+    });
+    return init;
+  });
+
+  const updateQuestionTags = (tagsKey: string, newTags: TagOption[]) => {
+    setQuestionTags({ ...questionTags, [tagsKey]: newTags });
+  };
+
+  const trigger = React.cloneElement(triggerButton, {
+    onClick: () => {
+      setOpenForm(true);
+    },
+  });
+
```

Comment:
It might be cleaner to let the client of this component to handle the onClick, we may want to add more features for the onClick that could differentiate for "creating a question" and "editing a form".

Reply:
okk I fixed it now the client can add more features to their button while it can still be used to open the form when passed into the QuestionForm